src/Microsoft.AspNetCore.OData/Batch/ODataBatchMiddleware.cs

```diff
@@ -34,13 +34,15 @@ public ODataBatchMiddleware(RequestDelegate next)
         /// <returns>A task that can be awaited.</returns>
         public async Task Invoke(HttpContext context)
         {
-            // Attempt to match the path to a bach route.
+            HttpRequest request = context.Request;
+            bool isPreFlight = HttpMethods.IsOptions(request.Method);
+           
+            // Attempt to match the path to a batch route.
             ODataBatchPathMapping batchMapping = context.RequestServices.GetRequiredService<ODataBatchPathMapping>();
 
-            string routeName;
-            if (batchMapping.TryGetRouteName(context, out routeName))
+            if (!isPreFlight && batchMapping.TryGetRouteName(context, out var routeName))
```

Comment:
```suggestion
            if (!isPreFlight && batchMapping.TryGetRouteName(context, out string routeName))
```
I however think that both `out string routeName` and `out var routeName` would be flagged by the Fx analyzer which is why in lots of places we declare the variable first before using it as an out parameter